Add header text align theme variable to stack cards

dhoeben opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->
I added a variable to hui-stack, which will impact `horizontal-stack`, `vertical-stack` and `grid`. This way, when defining a variable in the theme, users are able to align their title in the way they want. I also added a fallback, to align the title at the start, as it is now.

Info: https://developer.mozilla.org/en-US/docs/Web/CSS/text-align


## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

```yaml
theme:
  ha-card-header-text-align: center
```


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

dhoeben committed:
Update hui-stack-card.ts

Added variable

dhoeben committed:
Merge branch 'dev' into title-alignment-option

dhoeben committed:
Merge branch 'dev' into title-alignment-option

dhoeben committed:
Merge branch 'dev' into title-alignment-option

bramkragten commented on src/panels/lovelace/cards/hui-stack-card.ts:
```diff
@@ -92,6 +92,7 @@ export abstract class HuiStackCard<T extends StackCardConfig = StackCardConfig>
     return css`
       .card-header {
         color: var(--ha-card-header-color, var(--primary-text-color));
+        text-align: var(--ha-card-header-text-align, start);
```
The variable sounds like it is for `ha-card` but you implemented it just for stack cards. I would expect this to be at least also implemented in `ha-card`.

dhoeben replied:
Better like this?

dhoeben commented:
The variable is in line with the rest of the variable names used in the hui-stack. If you don't like the variable names, I can change all of them if you like!

bramkragten commented:
> The variable is in line with the rest of the variable names used in the hui-stack. If you don't like the variable names, I can change all of them if you like!

Thats is because it is using the variables of `ha-card`, because it uses the same style as `ha-card`. You should definitely not change those.

dhoeben commented:
So you are saying that you don't want to add the possibility to align titles of the vertical stack card, horizontal stack card and grid card?

bramkragten commented:
> So you are saying that you don't want to add the possibility to align titles of the vertical stack card, horizontal stack card and grid card?

No? I'm just saying you should add to more places if we are going to add it, to make it consistent.

dhoeben commented:
The difference is that with the hui-stack, the title is placed "outside" the card. And with other cards it is located "inside" the card. As titles are different for stacks then for cards.

dhoeben committed:
Merge branch 'dev' into title-alignment-option

dhoeben committed:
Update hui-stack-card.ts

Updated the variable, so that it would not be in line with the rest of the variables. In this way, the variable only works for hui-stack titles.

dhoeben committed:
Merge branch 'dev' into title-alignment-option

bramkragten commented on src/panels/lovelace/cards/hui-stack-card.ts:
```diff
@@ -92,6 +92,7 @@ export abstract class HuiStackCard<T extends StackCardConfig = StackCardConfig>
     return css`
       .card-header {
         color: var(--ha-card-header-color, var(--primary-text-color));
+        text-align: var(--hui-stack-title-text-align, start);
```
```suggestion
        text-align: var(--ha-stack-title-text-align, start);
```

dhoeben committed:
Update hui-stack-card.ts

Co-authored-by: Bram Kragten <mail@bramkragten.nl>

dhoeben committed:
Merge branch 'dev' into title-alignment-option

dhoeben committed:
Merge branch 'dev' into title-alignment-option
